platform_windows.cs

```diff
@@ -0,0 +1,184 @@
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Windows.Forms;
+using Microsoft.Toolkit.Forms.UI.Controls;
+
+[assembly: AssemblyTitle("WebApp")]
+[assembly: AssemblyDescription("")]
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("")]
+[assembly: AssemblyProduct("WebApp")]
+[assembly: AssemblyCopyright("")]
+[assembly: AssemblyTrademark("")]
+[assembly: AssemblyCulture("")]
+[assembly: ComVisible(false)]
+[assembly: Guid("91b0c988-0c36-44eb-9101-f6ef25f73dfa")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
+
+namespace WebApp
+{
+    public delegate void VoidCallback();
+    public delegate byte BoolCallback();
+    public delegate int Int32Callback();
+
+    public class PlatformWindows
+    {
+        public const int QuitResponseCancel = 0;
+        public const int QuitResponseNow = 1;
+        public const int QuitResponseLater = 2;
+
+        private readonly VoidCallback willFinishStartup;
+        private readonly VoidCallback didFinishStartup;
+        private readonly VoidCallback willActivate;
+        private readonly VoidCallback didActivate;
+        private readonly VoidCallback willDeactivate;
+        private readonly VoidCallback didDeactivate;
+        private readonly BoolCallback quitAfterLastWindowClosed;
+        private readonly Int32Callback checkQuit;
+
+        private RootWindow rootWindow;
+        private List<WebWindow> webWindows = new List<WebWindow>();
+        private MenuBar menuBar;
+
+        public PlatformWindows(IntPtr willFinishStartup, IntPtr didFinishStartup, IntPtr willActivate, IntPtr didActivate, IntPtr willDeactivate, IntPtr didDeactivate, IntPtr quitAfterLastWindowClosed, IntPtr checkQuit)
+        {
+            this.willFinishStartup = Marshal.GetDelegateForFunctionPointer<VoidCallback>(willFinishStartup);
+            this.didFinishStartup = Marshal.GetDelegateForFunctionPointer<VoidCallback>(didFinishStartup);
+            this.willActivate = Marshal.GetDelegateForFunctionPointer<VoidCallback>(willActivate);
+            this.didActivate = Marshal.GetDelegateForFunctionPointer<VoidCallback>(didActivate);
+            this.willDeactivate = Marshal.GetDelegateForFunctionPointer<VoidCallback>(willDeactivate);
+            this.didDeactivate = Marshal.GetDelegateForFunctionPointer<VoidCallback>(didDeactivate);
+            this.quitAfterLastWindowClosed = Marshal.GetDelegateForFunctionPointer<BoolCallback>(quitAfterLastWindowClosed);
+            this.checkQuit = Marshal.GetDelegateForFunctionPointer<Int32Callback>(checkQuit);
+        }
+
+        public void Start()
+        {
+            Application.EnableVisualStyles();
+            Application.SetCompatibleTextRenderingDefault(false);
+            rootWindow = new RootWindow(checkQuit);
+            willFinishStartup();
```

Comment:
Once this is called, it should be safe for the app developer to make calls to the library to create windows and other wise interact with things. Is that the case here?

Reply:
Yes, the test app does this and the things it does show up.